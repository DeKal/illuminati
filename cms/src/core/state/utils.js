export const toSuccess = actionName => `${actionName}Success`

export const toError = actionName => `${actionName}Error`

export const toRequest = actionName => `${actionName}Request`
