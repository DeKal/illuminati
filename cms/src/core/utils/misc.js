import { POSTS_PER_PAGE } from 'core/const/posts'

export const isPathMatchUrl = (pathName, url, isExact = false) => {
  if (url === '/' || pathName === '/' || isExact) {
    return pathName === url
  }
  return pathName.includes(url)
}

export const stringtifyTags = (tags, limits = 20) => {
  const tagsStr = tags.join(', ')
  if (tagsStr.length > limits) {
    return tagsStr.slice(0, limits) + '...'
  }
  return tagsStr
}

export const calcMaxRowsPerPage = postCount => {
  const postsPerPageOptions = []
  for (let index = 0; index < POSTS_PER_PAGE.length; index++) {
    postsPerPageOptions.push(POSTS_PER_PAGE[index])
    if (POSTS_PER_PAGE[index] > postCount) {
      break
    }
  }
  return postsPerPageOptions
}

export const arrayToAttributes = array =>
  array.reduce((object, attribute) => {
    return {
      ...object,
      [attribute]: true
    }
  }, {})

export const attributesToArray = obj =>
  Object.keys(obj).map(attribute => (obj[attribute] ? attribute : null))
