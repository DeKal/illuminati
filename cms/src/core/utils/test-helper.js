import React from 'react'
import { shallow, mount } from 'enzyme'
import Theme from 'core/theme/ThemeWrapper'

export const shallowWithTheme = component => {
  const wrapper = shallow(
    <Theme>
      <div data-test-id="theme-wrapper">{component}</div>
    </Theme>
  )
  return wrapper.find('[data-test-id="theme-wrapper"]').shallow()
}

export const mountWithTheme = component => {
  const wrapper = mount(
    <Theme>
      <div data-test-id="theme-wrapper">{component}</div>
    </Theme>
  )
  return wrapper.find('[data-test-id="theme-wrapper"]')
}
