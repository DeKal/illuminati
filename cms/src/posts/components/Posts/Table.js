import { withStyles } from '@material-ui/core/styles'
import Table from '@material-ui/core/Table'

const CustomTable = withStyles(theme => ({
  root: {
    minWidth: theme.postsTableWidth
  }
}))(Table)

export default CustomTable
