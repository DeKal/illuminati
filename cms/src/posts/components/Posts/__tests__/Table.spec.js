import * as React from 'react'
import { mountWithTheme } from 'core/utils/test-helper'
import Table from '../Table'

describe('Paper', () => {
  it('should render with no error', function() {
    const wrap = mountWithTheme(<Table />)
    expect(wrap).toMatchSnapshot()
  })
})
