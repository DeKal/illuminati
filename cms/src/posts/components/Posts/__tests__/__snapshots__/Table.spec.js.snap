// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paper should render with no error 1`] = `
<div
  data-test-id="theme-wrapper"
>
  <WithStyles(WithStyles(ForwardRef(Table)))>
    <WithStyles(ForwardRef(Table))
      classes={
        Object {
          "root": "WithStyles(ForwardRef(Table))-root-1",
        }
      }
    >
      <ForwardRef(Table)
        classes={
          Object {
            "root": "MuiTable-root WithStyles(ForwardRef(Table))-root-1",
            "stickyHeader": "MuiTable-stickyHeader",
          }
        }
      >
        <table
          className="MuiTable-root WithStyles(ForwardRef(Table))-root-1"
          role={null}
        />
      </ForwardRef(Table)>
    </WithStyles(ForwardRef(Table))>
  </WithStyles(WithStyles(ForwardRef(Table)))>
</div>
`;
